logging/src/lib.rs

```diff
@@ -12,6 +12,10 @@ pub fn init_logging<P: AsRef<std::path::Path>>(_log_file_path: Option<P>) {
     env_logger::init();
 }
 
+pub fn try_init_logging<P: AsRef<std::path::Path>>(_log_file_path: Option<P>) {
```

Comment:
Why?! Initializing logging should never fail.

Reply:
If someone else has already initialized the logger then trying to initialize it will fail. There is a clash when running the tests, if each initializes its own logger.

Reply:
Then this should be fixed either by running it in a singleton or fixing the logging library to have that singleton. We don't want a "try" because there's no way to distinguish between a failure and re-initialization.

Please remove the `try` variant, and make your tests solve this problem in a different way, with some kind of a call_once mechanism.

https://doc.rust-lang.org/std/sync/struct.Once.html

Reply:
I'm also happy to have this used in the logging library itself. You can define that Once variable in logging itself.

Reply:
This was fixed, btw. It never fails now and can be initialized multiple times.

Reply:
Removed the `try_init_logging` function.